src/controller.rs

```diff
@@ -103,619 +193,733 @@ enum JobOutcome {
+            AppInstanceLikeResources::AppInstance(_) => {
+                let api: Api<AppInstance> = Api::namespaced(ctx.client.clone(), ns);
+                Ok(api
+                    .get_status(&self.instance.name_any())
+                    .await?
+                    .status
+                    .clone()
+                    .unwrap_or_default())
+            }
+            AppInstanceLikeResources::ConfigMap(_) => {
+                let api: Api<ConfigMap> = Api::namespaced(ctx.client.clone(), ns);
+                let data = api.get(&self.instance.name_any()).await?.data;
+                if let Some(data) = data {
+                    let status = data.get("status");
+                    if let Some(status) = status {
+                        serde_json::from_str::<AppInstanceStatus>(status.as_str())
+                            .map_err(Error::InvalidConfigMapStatus)
+                    } else {
+                        Ok(AppInstanceStatus::default())
+                    }
+                } else {
+                    Ok(AppInstanceStatus::default())
                 }
             }
+        }
+    }
 
-            per_container_logs
-                .entry(container_name.clone())
-                .and_modify(|e: &mut String| e.push_str(&logs))
-                .or_insert(logs);
+    async fn update_status(&self, ctx: &Context, status: AppInstanceStatus) -> Result<()> {
+        let ns = &self.instance.namespace().ok_or(Error::NamespaceRequired)?;
+
+        match self.original {
+            AppInstanceLikeResources::AppInstance(_) => {
+                let app_instance_api: Api<AppInstance> = Api::namespaced(ctx.client.clone(), ns);
+
+                let app_instance_patch = AppInstance {
+                    metadata: Default::default(),
+                    spec: Default::default(),
+                    status: Some(status),
+                };
+                app_instance_api
+                    .patch_status(
+                        &self.instance.name_any(),
+                        &patch_params(),
+                        &Patch::Apply(&app_instance_patch),
+                    )
+                    .await?;
+            }
+            AppInstanceLikeResources::ConfigMap(_) => {
+                let config_map_api: Api<ConfigMap> = Api::namespaced(ctx.client.clone(), ns);
+                let status_string =
+                    serde_json::to_string(&status).map_err(Error::InvalidConfigMapStatus)?;
+                let patch = serde_json::json!({
+                    "apiVersion": "v1",
+                    "kind": "ConfigMap",
+                    "data": {
+                        "status": status_string,
+                    }
+                });
+                config_map_api
+                    .patch(
+                        &self.instance.name_any(),
+                        &patch_params(),
+                        &Patch::Apply(&patch),
+                    )
+                    .await?;
```

Comment:
I think a quick-win, building off what I've seen the `cluster-autoscaler` do for its `ConfigMap` updates, is to use a timestamp as well. So you know when it last ran (or simply mutating an annotation too). E.g. 

```
kubectl get cm -n cluster-autoscaler cluster-autoscaler-status -o yaml
apiVersion: v1
data:
  status: |+
    Cluster-autoscaler status at 2023-12-05 15:40:55.94549564 +0000 UTC:
    Cluster-wide:
      Health:      Healthy (ready=35 unready=0 (resourceUnready=0) notStarted=0 longNotStarted=0 registered=35 longUnregistered=0)
                   LastProbeTime:      2023-12-05 15:40:55.570229638 +0000 UTC m=+15102052.903577438
                   LastTransitionTime: 2023-06-13 20:40:31.747397597 +0000 UTC m=+29.080745330
    
   ...lots of other node info below this
kind: ConfigMap
metadata:
  annotations:
    cluster-autoscaler.kubernetes.io/last-updated: 2023-12-05 15:40:55.94549564 +0000 UTC
  name: cluster-autoscaler-status
  namespace: cluster-autoscaler
```

Reply:
So I think this is already pretty much handled by the `lastTransitionTime` on the `AppInstanceCondition`, or am I misunderstanding your request?

Reply:
Actually that is a fair point, I completely forgot about that. It should be handled in that case already :100:

I can't really think of a reason why we'd want to duplicate that. As far as I know the `lastTransitionTime` is automatically updated on status patch, which should be more than sufficient.

Reply:
I just thought actually, I meant this comment from the "configmap mode" perspective, although if the `lastTransitionTime` is placed into the condition _within_ the `AppInstance` that is shown in the `ConfigMap` itself then it's all good.

Maybe I'm misunderstanding what you're saying now that I think about it :smile: